coderd/database/queries.sql.go

```diff
@@ -12499,30 +12501,31 @@ type GetWorkspacesParams struct {
-	TemplateVersionID      uuid.UUID           `db:"template_version_id" json:"template_version_id"`
-	TemplateVersionName    sql.NullString      `db:"template_version_name" json:"template_version_name"`
-	Username               string              `db:"username" json:"username"`
-	LatestBuildCompletedAt sql.NullTime        `db:"latest_build_completed_at" json:"latest_build_completed_at"`
-	LatestBuildCanceledAt  sql.NullTime        `db:"latest_build_canceled_at" json:"latest_build_canceled_at"`
-	LatestBuildError       sql.NullString      `db:"latest_build_error" json:"latest_build_error"`
-	LatestBuildTransition  WorkspaceTransition `db:"latest_build_transition" json:"latest_build_transition"`
-	Count                  int64               `db:"count" json:"count"`
+	ID                     uuid.UUID                `db:"id" json:"id"`
+	CreatedAt              time.Time                `db:"created_at" json:"created_at"`
+	UpdatedAt              time.Time                `db:"updated_at" json:"updated_at"`
+	OwnerID                uuid.UUID                `db:"owner_id" json:"owner_id"`
+	OrganizationID         uuid.UUID                `db:"organization_id" json:"organization_id"`
+	TemplateID             uuid.UUID                `db:"template_id" json:"template_id"`
+	Deleted                bool                     `db:"deleted" json:"deleted"`
+	Name                   string                   `db:"name" json:"name"`
+	AutostartSchedule      sql.NullString           `db:"autostart_schedule" json:"autostart_schedule"`
+	Ttl                    sql.NullInt64            `db:"ttl" json:"ttl"`
+	LastUsedAt             time.Time                `db:"last_used_at" json:"last_used_at"`
+	DormantAt              sql.NullTime             `db:"dormant_at" json:"dormant_at"`
+	DeletingAt             sql.NullTime             `db:"deleting_at" json:"deleting_at"`
+	AutomaticUpdates       AutomaticUpdates         `db:"automatic_updates" json:"automatic_updates"`
+	Favorite               bool                     `db:"favorite" json:"favorite"`
+	TemplateName           string                   `db:"template_name" json:"template_name"`
+	TemplateVersionID      uuid.UUID                `db:"template_version_id" json:"template_version_id"`
+	TemplateVersionName    sql.NullString           `db:"template_version_name" json:"template_version_name"`
+	Username               string                   `db:"username" json:"username"`
+	LatestBuildCompletedAt sql.NullTime             `db:"latest_build_completed_at" json:"latest_build_completed_at"`
+	LatestBuildCanceledAt  sql.NullTime             `db:"latest_build_canceled_at" json:"latest_build_canceled_at"`
+	LatestBuildError       sql.NullString           `db:"latest_build_error" json:"latest_build_error"`
+	LatestBuildTransition  WorkspaceTransition      `db:"latest_build_transition" json:"latest_build_transition"`
+	LatestBuildStatus      NullProvisionerJobStatus `db:"latest_build_status" json:"latest_build_status"`
```

Comment:
Is there a way to make this be non-nullable in the query?

Reply:
Afraid not, looks like the `latest_build` attributes are the result of a left join: https://github.com/coder/coder/blob/843d65035f0c1c436e24ef2ac5121b7c5b58784d/coderd/database/queries.sql.go#L12172-L12218

Reply:
I suppose by definition a workspace cannot exist without a build...

I see in `workspace_builds`:

```sql
create table if not exists public.workspace_builds
(
    id                  uuid                                                                                not null
        primary key,
    created_at          timestamp with time zone                                                            not null,
    updated_at          timestamp with time zone                                                            not null,
    workspace_id        uuid                                                                                not null
        references public.workspaces
            on delete cascade,
...
```

The workspace foreign key is not null, so I think this left join may be inappropriate?

Reply:
Hhmm. I changed it and ran `make gen` but it's still being defined as a nullable field.
Also, one of its sibling fields `latest_build_transition` is not nullable and is derived from the same join, so actually all of the above about the join was ostensibly was invalid.

I'll need to dig a little further.

Reply:
Ah! I was basically on the correct track!

`provisioner_jobs` was being left joined as well, and since that also has a non-nullable foreign key with `workspace_builds.job_id`, it is safe to use an inner join. I can actually leave the outer lateral join as it is but just change this to correct the generated code.

Reply:
@johnstcn see [b118044](https://github.com/coder/coder/pull/12762/commits/b118044549d20a76ffe6c1a45a440a64e3f7f9a9)